decorator.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
```

Comment:
[Optional] It is usually a good practice to include a message that will further explain the error to the user. For example:
`raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}`